pool/src/transaction_pool.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018 The Grin Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A minimal (EXPERIMENTAL) transaction pool implementation
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use core::core::hash::{Hash, Hashed};
+use core::core::id::ShortIdentifiable;
+use core::core::transaction;
+use core::core::{Block, Committed, CompactBlock, Transaction};
+use keychain::BlindingFactor;
+use pool::Pool;
+use types::*;
+use util::{secp_static, static_secp_instance};
+use util::secp::pedersen::Commitment;
+use util::LOGGER;
+
+/// A minimal (EXPERIMENTAL) transaction pool implementation
+pub struct TransactionPool<T> {
+	/// Pool Config
+	pub config: PoolConfig,
+
+	/// Our transaction pool.
+	pub pool: Pool<T>,
+	/// TODO - Our stem transaction pool (just another instance of a pool?)
+	pub stem_pool: Pool<T>,
```

Comment:
We can use the same pool instance or simplify it by creating another instance ```Stempool``` (which implement the ```Pool```) which is basically the same but with ```time_stem_transactions``` (to track the transactions). 
Also ```stempool``` and not ```stem_pool``` :) https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-May/015956.html.